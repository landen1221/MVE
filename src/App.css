/* FIXME: Delete at some point */
/* body {
  background-color: darkslateblue;
} */

a {
  text-decoration: none;
}

.App {
  text-align: center;
  margin-top: 5em;
  margin-bottom: 10em;
  font-size: Roboto, Helvetica, Arial, sans-serif;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}
