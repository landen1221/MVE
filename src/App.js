import "./App.css";
import StoriesSection from "./javascript/StoriesSection";
import StoryForm from "./javascript/StoryForm";
import { Route, Switch, Redirect } from "react-router-dom";
import Navbar from "./javascript/Navbar";
import Header from "./javascript/Header";

function App() {
  return (
    <div className="App">
      <Navbar />

      <Switch>
        <Route exact path="/covid">
          <StoriesSection vaccine="COVID" />
        </Route>
        <Route exact path="/vaccine/pfizer">
          <StoriesSection vaccine="Pfizer" />
        </Route>
        <Route exact path="/vaccine/moderna">
          <StoriesSection vaccine="Moderna" />
        </Route>
        <Route exact path="/vaccine/johnsonandjohnson">
          <StoriesSection vaccine="Johnson & Johnson" />
        </Route>
        <Route exact path="/vaccine/astrazeneca">
          <StoriesSection vaccine="AstraZeneca" />
        </Route>
        <Route exact path="/add-story">
          <StoryForm />
        </Route>
        <Redirect from="/" to="/covid" />
      </Switch>
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
    </div>
  );
}

export default App;
