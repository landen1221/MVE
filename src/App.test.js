import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
});

// placeholder
// test("1 === 1", () => {
//   expect(1).toBe(1);
// });
