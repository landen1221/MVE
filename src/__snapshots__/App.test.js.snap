// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Navbar undefined"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <a
            href="/"
          >
            <span
              class="MuiTypography-root MuiTypography-noWrap"
              id="logo"
            >
              MyVaccineExperience.org
            </span>
          </a>
          <div
            class="makeStyles-search-12"
          >
            <div
              class="makeStyles-searchIcon-13"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root makeStyles-inputRoot-14"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input makeStyles-inputInput-15"
                name="search"
                placeholder="Search..."
                type="text"
                value=""
              />
            </div>
          </div>
          <a
            href="/add-story"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add Story
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
          <div />
          <div />
          <div />
        </div>
      </header>
    </div>
    <div
      class="Header"
    >
      <h2>
        Real stories from real people to help you make an informed decision!
      </h2>
      <div
        id="ButtonSection"
      >
        <a
          href="/covid"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            id="VaccineButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              COVID
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          href="/vaccine/johnsonandjohnson"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            id="VaccineButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Johnson & Johnson
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          href="/vaccine/astrazeneca"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            id="VaccineButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              AstraZeneca
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          href="/vaccine/pfizer"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            id="VaccineButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Pfizer
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          href="/vaccine/moderna"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            id="VaccineButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Moderna
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
    <div
      class="makeStyles-root-16"
    >
      <div
        class="StoriesSection"
      >
        <h3>
          Stories about 
          <i>
            COVID
          </i>
        </h3>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root StoriesSection-stories MuiGrid-item MuiGrid-grid-xs-9"
          >
            <div
              class="SortBox undefined"
            >
              <div
                class="MuiPaper-root makeStyles-greyPaper-17 MuiPaper-elevation2 MuiPaper-rounded"
              >
                <h4
                  id="filter"
                >
                  Filters: 
                </h4>
                <form>
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="select-gender"
                      id="select-gender-label"
                    >
                      Gender
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    >
                      <div
                        aria-haspopup="listbox"
                        aria-labelledby="select-gender-label select-gender"
                        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                        id="select-gender"
                        role="button"
                        tabindex="0"
                      >
                        <span>
                          ​
                        </span>
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput"
                        data-testid="gender"
                        name="gender"
                        tabindex="-1"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-18 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-20"
                        >
                          <span>
                            Gender
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="select-satisfied"
                      id="select-satisfied-label"
                    >
                      Intensity
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    >
                      <div
                        aria-haspopup="listbox"
                        aria-labelledby="select-satisfied-label select-satisfied"
                        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                        id="select-satisfied"
                        role="button"
                        tabindex="0"
                      >
                        <span>
                          ​
                        </span>
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput"
                        name="satisfied"
                        tabindex="-1"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-18 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-20"
                        >
                          <span>
                            Intensity
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                    >
                      Min. Age
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        name="ageMin"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-18 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-20"
                        >
                          <span>
                            Min. Age
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                    >
                      Max Age
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        name="ageMax"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-18 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-20"
                        >
                          <span>
                            Max Age
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
                    id="clear"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      clear filters
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
            </div>
            <div
              class="Stories makeStyles-root-22"
            >
              <i
                id="no-stories"
              >
                <br />
                no stories available
              </i>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div
              class="Stats"
            >
              <h4>
                Satified w/ Vaccine:
              </h4>
              <hr />
              <p
                id="intensity-helper"
              >
                (From 
                <u />
                 users)
              </p>
              <p
                class="satisfied"
              >
                <b>
                  J&J: 
                </b>
                0.5%
              </p>
              <p
                class="satisfied"
              >
                <b>
                  AstraZeneca: 
                </b>
                1%
              </p>
              <p
                class="satisfied"
              >
                <b>
                  Pfizer: 
                </b>
                1%
              </p>
              <p
                class="satisfied"
              >
                <b>
                  Moderna: 
                </b>
                0.67%
              </p>
            </div>
            <br />
            <div
              class="Stats"
            >
              <h4>
                COVID Intensity
              </h4>
              <hr />
              <p
                id="intensity-helper"
              >
                (Out of 
                <u>
                  2
                </u>
                 cases)
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
